lib/firmware_app.dart

```diff
@@ -37,17 +37,29 @@ class _FirmwareAppState extends State<FirmwareApp> {
   @override
   void initState() {
     super.initState();
-    context.read<FwupdNotifier>().init();
+    final fwupdNotifier = context.read<FwupdNotifier>();
     final store = context.read<DeviceStore>();
+    final gtkNotifier = getService<GtkApplicationNotifier>();
+
+    fwupdNotifier
+      ..init()
+      ..registerErrorListener(_showError)
+      ..registerConfirmationListener(_getConfirmation)
+      ..registerDeviceRequestListener(_showRequest);
+
     store.init();
-    final notifier = getService<GtkApplicationNotifier>();
-    notifier.addCommandLineListener((args) {
+    gtkNotifier.addCommandLineListener((args) {
```

Comment:
we may have a similar issue here that the listener is never removed. could this closure be a helper method so we can remove the listener in dispose()?

Reply:
:+1: